velox/functions/prestosql/window/AggregateWindow.cpp

```diff
@@ -216,13 +250,17 @@ class AggregateWindowFunction : public exec::WindowFunction {
     // incrementally aggregate over the new rows seen in the frame between
     // the previous and current row.
     for (int i = 0; i < numRows; i++) {
-      auto currentFrameEnd = rawFrameEnds[i] - startFrame + 1;
-      if (currentFrameEnd > prevFrameEnd) {
-        computeAggregate(rows, prevFrameEnd, currentFrameEnd);
+      if (isEmptyFrame(rawFrameStarts[i], rawFrameEnds[i])) {
```

Comment:
Shouldn't empty frame never occur in the incremental Aggregation case ?

Reply:
Thanks for pointing it out, this is needed for cases like "rows between unbounded preceding and k preceding" where empty frame can occur.

Reply:
I don't entirely follow the optimizations for empty frames. Lets discuss in person.